test/integration/preemption_toleration_test.go

```diff
@@ -0,0 +1,307 @@
+			preemptor: makePod(
+				schedulertesting.MakePod().Name("p").Container(pauseImage).ZeroTerminationGracePeriod().Priority(testPriority + 9),
+			).ResourceRequests(podRequest).Obj(),
+			waitPreemptorCreation: 30 * time.Second, // FIXME(everpeace): how to simulate forever??
+			canTolerate:           true,
+		},
+		{
+			name: "when preemptor's priority < MinimumPreemptablePriority(=testPriority+10), it can tolerate the preemption if victimCandidate is scheduled within TolerationSeconds",
+			priorityClass: makeTestPriorityClass(testPriorityClassName, testPriority, map[string]string{
+				preemptiontoleration.AnnotationKeyMinimumPreemptablePriority: fmt.Sprintf("%d", testPriority+10),
+				preemptiontoleration.AnnotationKeyTolerationSeconds:          fmt.Sprintf("%d", 30),
+			}),
+			preemptor: makePod(
+				schedulertesting.MakePod().Name("p").Container(pauseImage).ZeroTerminationGracePeriod().Priority(testPriority + 5),
+			).ResourceRequests(podRequest).Obj(),
+			canTolerate: true,
+		},
+		{
+			name: "when preemptor's priority < MinimumPreemptablePriority(=testPriority+10), it can NOT tolerate the preemption after TolerationSeconds elapsed from victimCandidate being scheduled",
+			priorityClass: makeTestPriorityClass(testPriorityClassName, testPriority, map[string]string{
+				preemptiontoleration.AnnotationKeyMinimumPreemptablePriority: fmt.Sprintf("%d", testPriority+10),
+				preemptiontoleration.AnnotationKeyTolerationSeconds:          fmt.Sprintf("%d", 5),
+			}),
+			preemptor: makePod(
+				schedulertesting.MakePod().Name("p").Container(pauseImage).ZeroTerminationGracePeriod().Priority(testPriority + 5),
+			).ResourceRequests(podRequest).Obj(),
+			waitPreemptorCreation: 10 * time.Second,
+			canTolerate:           false,
+		},
+		{
+			name: "when unparsable preemption toleration policy, victim should be preempted if preemptor's preemption policy is PreemptLowerPriority",
+			priorityClass: makeTestPriorityClass(testPriorityClassName, testPriority, map[string]string{
+				preemptiontoleration.AnnotationKeyMinimumPreemptablePriority: "a",
+			}),
+			preemptor: makePod(
+				schedulertesting.MakePod().Name("p").PreemptionPolicy(v1.PreemptLowerPriority).Container(pauseImage).ZeroTerminationGracePeriod().Priority(testPriority + 1),
+			).ResourceRequests(podRequest).Obj(),
+			canTolerate: false,
+		},
+		{
+			name: "when unparsable preemption toleration policy, victim should not be preempted if preemptor's preemption policy is PreemptNone",
+			priorityClass: makeTestPriorityClass(testPriorityClassName, testPriority, map[string]string{
+				preemptiontoleration.AnnotationKeyMinimumPreemptablePriority: "a",
+			}),
+			preemptor: makePod(
+				schedulertesting.MakePod().Name("p").PreemptionPolicy(v1.PreemptNever).Container(pauseImage).ZeroTerminationGracePeriod().Priority(testPriority + 1),
+			).ResourceRequests(podRequest).Obj(),
+			waitPreemptorCreation: 30 * time.Second, // FIXME(everpeace): how to simulate forever??
```

Comment:
You may inject a fake clock to simulate the time elapsing by `clock.Step()`. (search for `clock.Step()` in k/k for references).

Reply:
I think injecting a fake clock is difficult because this is the integration test.  We need some way to share a fake clock with the plugin (running on the scheduler) and the test.  Is this possible??

Reply:
I think injecting a fake clock is difficult because this is the integration test.  We need some way to share a fake clock with the plugin (running on the scheduler) and the test.  Is this possible??